server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -225,11 +243,21 @@ ActionListener<SearchResponse> transformResponseListener(
             final SearchResponseProcessor processor = searchResponseProcessors.get(i);
 
             responseListener = ActionListener.wrap(r -> {
+                ProcessorExecutionDetail detail = new ProcessorExecutionDetail(processor.getType());
+                if (request.source().verbosePipeline()) {
+                    detail.addInput(Arrays.asList(r.getHits().deepCopy().getHits()));
+                }
                 beforeResponseProcessor(processor);
                 final long start = relativeTimeSupplier.getAsLong();
                 processor.processResponseAsync(request, r, requestContext, ActionListener.wrap(rr -> {
                     long took = TimeUnit.NANOSECONDS.toMillis(relativeTimeSupplier.getAsLong() - start);
                     afterResponseProcessor(processor, took);
+                    if (request.source().verbosePipeline()) {
+                        detail.addOutput(Arrays.asList(rr.getHits().deepCopy().getHits()));
```

Comment:
Can you help me understand why this deepCopy is needed?

Reply:
When the Object receives this list, it holds a reference to the original SearchHit objects. Without a deepCopy, any modifications to the output will also reflect in the input, as they share the same reference.

By using deepCopy, we ensure that the input and output hold independent values, allowing us to clearly see the differences between the two at each step of the processor.